File: app/src/androidTest/java/com/juawapps/newsspread/di/TestActivityModule.java
package com.juawapps.newsspread.di;

import com.juawapps.newsspread.di.news.NewsFragmentBuildersModule;
import com.juawapps.newsspread.test.SingleFragmentActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;


@SuppressWarnings("unused")
@Module
abstract class TestActivityModule {

    @ContributesAndroidInjector(modules = {NewsFragmentBuildersModule.class})
    abstract SingleFragmentActivity bindMainActivity();
}

File: app/src/androidTest/java/com/juawapps/newsspread/di/TestActivityModule.kt
package com.juawapps.newsspread.di

import com.juawapps.newsspread.di.news.NewsFragmentBuildersModule
import com.juawapps.newsspread.test.SingleFragmentActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector


@Module
internal abstract class TestActivityModule {

    @ContributesAndroidInjector(modules = [NewsFragmentBuildersModule::class])
    internal abstract fun bindMainActivity(): SingleFragmentActivity
}